download/download.go

```diff
@@ -129,8 +139,6 @@ func (d *Downloader) oneShot(ctx context.Context) error {
 		d.f(ctx, Update{ETag: etag, Bundle: b, Error: err, Metrics: m})
 	}
 
-	d.etag = etag
```

Comment:
Just thinking out loud a little bit. I'm wondering if we are maybe leaking too much of the "downloader" details out by making the clients have to set the etag. From a high level I like that we can create a downloader and it will just like do the right thing with regards to cache controls. In this particular case we are kind of working around issues with bundle dependencies by manually setting them to trigger re-downloading and trying to re-activate the bundle.

I guess maybe what I'm leaning towards is having the downloader still automatically set the etag, but maybe just do it before calling the listener so they can clear it in their callback if they need to.

I'm wondering too if we maybe want to just have a like `ClearCache()` or `Reset()` or something a little bit more opaque  instead of making the bundle and discovery plugins know that setting the etag to "" means it will download it again. I'm less sure about this direction though..  WDYT?

Reply:
@patrick-east  I like the idea of setting the etag in downloader and clearing it in the client callback if some error occurs. I was thinking if the callback could be updated to return an error and then the downloader would set/reset the etag accordingly. But your approach seems good. 

So in the downloader we would be adding a `ClearCache()` api that would basically set the etag to `""` , correct ?

Reply:
Yea, I think so.

Reply:
@patrick-east  updated the code as per the above discussion.